### PR TITLE
ci: Bump Ubuntu and kernel versions for CI builds

### PR DESCRIPTION
**Pull request by aleksamagicka**
Just noticed that the kernel versions the CI uses are fairly old. Bump them up to current ones, including the Ubuntu version.

---

Shouldn't break anything, I hope? Or should I leave the 5.x ones in and only add these new ones?

### CODE REVIEW

**Comment by amezin** [comment]
I think for kernel versions we should have at least:

- `master`
- oldest version that we support (I guess it's 5.10)

5.15 was just the latest stable version at the time the pipeline was added. Maybe it would make sense to track the latest stable release (for the case when `master` is broken, for example).

For Ubuntu versions - maybe just set them to `ubuntu-latest` everywhere and forget?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,15 +9,15 @@ jobs:
       matrix:
         kernel_version:
           - master
-          - v5.15
-          - v5.10
```
Comment (amezin) [inline .github/workflows/build.yml:13]: If we still support 5.10 - please keep it here.

What's the "oldest supported version" of the kernel?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -9,15 +9,15 @@ jobs:
...
+          - v6.5
+          - v6.1
         kconfig_pm:
           - "-e CONFIG_PM"
           - "-d CONFIG_PM"
         kconfig_debug_fs:
           - "-e CONFIG_DEBUG_FS"
           - "-d CONFIG_DEBUG_FS"
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```
Comment (amezin) [inline .github/workflows/build.yml:20]: Maybe just `ubuntu-latest`?

Also, some other workflow files have `ubuntu-20.04` in them too

**Review by amezin — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (aleksamagicka) [reply .github/workflows/build.yml:20]: `ubuntu-latest` SGTM.

> Also, some other workflow files have `ubuntu-20.04` in them too

I only see `build.yml` using 20.04, which ones also have it?

File: .github/workflows/build.yml (hunk 1 above)
Comment (aleksamagicka) [reply .github/workflows/build.yml:13]: Will re-add 5.10.

> What's the "oldest supported version" of the kernel?

Right now, it's [4.14](https://kernel.org/category/releases.html). Ubuntu 20.04 is on 5.4, with updates to 5.8. I don't see a point going below that, although CI is free here.

Comment (amezin) [reply .github/workflows/build.yml:13]: I meant oldest supported by liquidtux. Not sure if we ever discussed that. @jonasmalacofilho 

Personally I think anything older than "latest Ubuntu LTS" (which is now 22.04) is likely not necessary

File: .github/workflows/build.yml (hunk 2 above)
Comment (amezin) [reply .github/workflows/build.yml:20]: Oops, my bad. They have `ubuntu-22.04`. But if we're switching to `latest` here, it makes sense to switch everywhere at once.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jonasmalacofilho) [reply .github/workflows/build.yml:13]: > I meant oldest supported by liquidtux. Not sure if we ever discussed that. @jonasmalacofilho

I don't have an opinion on this.

But the kernel from the "latest Ubuntu LTS" makes sense, at least as a starting point. And things on hwmon and hid move fairly slowly.

Comment (aleksamagicka) [reply .github/workflows/build.yml:13]: I've now changed it to 6.2, as that's what's in the current version of 22.04. Should I add anything else?

**Review by amezin — APPROVED** [review]

**Pull request merged**